### ISSUE TITLE
Can support distributed deploy?

### ISSUE THREAD

**Issue by greyireland** [jamesdbloom/mockserver#665]
Hi ,I want use mockserver for load testing.I wonder single node maybe have some problem.
Can it afford 50w/s QPS?

**Comment by jamesdbloom** [comment]
I will add a page to the website with performance figures soon, but the current performance test shows the figures below.  These were recorded with 4 expectations in MockServer where the request matched the 3rd expectation, all figures are in **milliseconds**.

|   req/s  |   reqs | Avg | Min | Max | Median | 50% | 66% | 75% | 80% | 90% | 95% | 98% |99% | 100%
:----------|:-------|:----|:----|:----|:-------|:----|:----|:----|:----|:----|:----|:----|:---|:----
|   50     |   2476 |   2 |   1 |  12 |      2 |   2 |   3 |   3 |   3 |   3 |   4 |   5 |  5 |   12
|   99     |   4891 |   4 |   0 |  18 |      3 |   3 |   4 |   5 |   5 |   6 |   7 |   8 |  9 |   18
|  496     |  22271 |   7 |   0 |  65 |      6 |   6 |   7 |   9 |  10 |  12 |  15 |  19 | 22 |   65	
|  995     | 106830 |  20 |   0 | 236 |      6 |   6 |  13 |  19 |  25 |  57 |  97 | 140 |160 |  240
| 1243     | 135671 | 113 |   0 | 434 |    110 | 110 | 140 | 160 | 170 | 210 | 250 | 290 |300 |  430

In summary, MockServer can easily handle 50 TPS with a p99 of 5ms and can scale up to 1250 TPS with a p99 of 300ms.

**Comment by jamesdbloom** [comment]
There are two upcoming features that may provide the mechanism you want to support some basic distribution, one is to support expectation persistence (to a local file system file) https://trello.com/c/qOgJgBee/26-persistence-expectations the other is to reload expectations when an expectation initialisation file changes https://trello.com/c/MN3KRvau/32-hot-load-expectation-initialization-file.  If the local file system was shared between multiple VMs then these two features would provide the mechanisms to support distribution, however, the performance wouldn't be good if the expectations were changing rapidly due to the overhead of the file system. 

I'm going to close this ticket as clustering/distribution is not on the roadmap due to the complexity and performance overhead that would introduce.
